.github/workflows/check-go-validation.yml

```
@@ -0,0 +1,26 @@
+name: "[Check] go validation"
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
```

> The action `golangci/golangci-lint-action` seems official, so it should be good.

If you want to be on the safe side, you can also limit the GITHUB_TOKEN permissions, see https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs and https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token.

This action only ever needs read access to do a `git checkout`, but by default the `GITHUB_TOKEN` has a lot of permissions.

Even locking your `uses: golangci/golangci-lint-action@<specific-tag-here>` to a specific tag isn't good enough, since in git, you can easily overwrite tags.

```suggestion
permissions:
  contents: read
```

---

`GITHUB_TOKEN` is set to read-only in repository settings

---

lets add this as it doesn't hurt to have it there :)